compiler/dyno/test/uast/testStringify.cpp

```diff
@@ -436,9 +455,12 @@ static void test4(Parser* parser) {
 
 static void test5(Parser* parser) {
   // unsure how best to set this up for testing
+  // probably need to parse an input to a node, and send that node into a function
```

Comment:
I think the best thing to do is just to map a string to a file using `setFileContents`. Then you parse and syntaxify it, and can print that to a string stream. Then you just assert that the stream contents and the file contents are the same.

```cpp
auto path = UniqueString::get(context, "test5.chpl");
std::string contents = R""""(
module Foobar {}
)"""";
setFileText(context, path, contents);
auto& br = parseFile(context, path);
assert(!br.numErrors());
auto mod = br.singleModule();
assert(mod);
std::ostringstream ss;
printChapelSyntax(ss, mod);
assert(ss.str() == fileText(context, path).text());
```

Ah, but I just read the macros and you are already doing something similar. I'm not sure what the problem is here, feel free to poke me to talk about it!